docs/fr/flutterfirebaselogintutorial.md

```diff
@@ -0,0 +1,1965 @@
+
+## Register Button (Bouton d'inscription)
+
+Créons `register/register_button.dart` et commençons.
+
+```dart
+import 'package:flutter/material.dart';
+
+class RegisterButton extends StatelessWidget {
+  final VoidCallback _onPressed;
+
+  RegisterButton({Key key, VoidCallback onPressed})
+      : _onPressed = onPressed,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return RaisedButton(
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(30.0),
+      ),
+      onPressed: _onPressed,
+      child: Text('Register'),
+    );
+  }
+}
+```
+C'est très similaire à l'installation faite pour `LoginButton`, le `RegisterButton` a du code pour changer son style et il expose une fonction  `VoidCallback` pour que l'on puisse gérer le moment où l'utilisateur va appuyer sur le bouton de le Widget parent.
+
+Tout ce qui nous reste à faire est d'actualiser notre widget `App` dans `main.dart` pour afficher le `LoginScreen` si le `AuthenticationState` est `Unauthenticated`.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:bloc/bloc.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_firebase_login/authentication_bloc/authentication_bloc.dart';
+import 'package:flutter_firebase_login/user_repository.dart';
+import 'package:flutter_firebase_login/home_screen.dart';
+import 'package:flutter_firebase_login/login/login.dart';
+import 'package:flutter_firebase_login/splash_screen.dart';
+import 'package:flutter_firebase_login/simple_bloc_delegate.dart';
+
+void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+  BlocSupervisor.delegate = SimpleBlocDelegate();
+  final UserRepository userRepository = UserRepository();
+  runApp(
+    BlocProvider(
+      create: (context) => AuthenticationBloc(userRepository: userRepository)
+        ..add(AppStarted()),
+      child: App(userRepository: userRepository),
+    ),
+  );
+}
+
+class App extends StatelessWidget {
+  final UserRepository _userRepository;
+
+  App({Key key, @required UserRepository userRepository})
+      : assert(userRepository != null),
+        _userRepository = userRepository,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: BlocBuilder<AuthenticationBloc, AuthenticationState>(
+        builder: (context, state) {
+          if (state is Uninitialized) {
+            return SplashScreen();
+          }
+          if (state is Unauthenticated) {
+            return LoginScreen(userRepository: _userRepository);
+          }
+          if (state is Authenticated) {
+            return HomeScreen(name: state.displayName);
+          }
+        },
+      ),
+    );
+  }
+}
+```
+
+Maintenant nous avons un formulaire de connexion assez robuste utilisant Firebase and nous avons notre couche de présentation qui est séparé de notre couche de business logic tout cela en utilisant la librairie Bloc.
```

Comment:
and nous avons -> et nous avons
qui est séparé -> qui est séparée
la librairie -> la bibliothèque